server/plugin/utils.go

```diff
@@ -457,15 +457,29 @@ func (p *Plugin) deleteSubscription(subscription *serializers.SubscriptionDetail
 		return http.StatusInternalServerError, deleteErr
 	}
 
+	if deleteErr := p.Store.DeleteSubscriptionChannelID(subscription.SubscriptionID); deleteErr != nil {
+		return http.StatusInternalServerError, deleteErr
+	}
+
 	return http.StatusOK, nil
 }
 
-func (p *Plugin) VerifyWebhookSecret(received string) (status int, err error) {
-	if p.getConfiguration().WebhookSecret != received {
-		return http.StatusForbidden, errors.New(constants.ErrorUnauthorisedSubscriptionsWebhookRequest)
+func (p *Plugin) VerifySubscriptionWebhookSecretAndGetChannelID(subscriptionID, uniqueWebhookSecret string) (string, int, error) {
+	subscriptionWebhookSecret, err := p.Store.GetSubscriptionChannelID(subscriptionID)
+	if err != nil {
+		return "", http.StatusInternalServerError, err
 	}
 
-	return 0, nil
+	if subscriptionWebhookSecret != nil {
```

Comment:
we should try to do less things inside if else statements if possible like 
```suggestion
	if subscriptionWebhookSecret == nil { 
	return "", http.StatusForbidden, errors.New(constants.ErrorUnauthorisedSubscriptionsWebhookRequest)
	}
	
			webhookSecret := *subscriptionWebhookSecret
		if webhookSecret[uniqueWebhookSecret] == "" {
			return "", http.StatusForbidden, errors.New(constants.ErrorUnauthorisedSubscriptionsWebhookRequest)
		}

		return webhookSecret[uniqueWebhookSecret], http.statusOK, nil
	
	
```